agora-drep/Agora/Proxy.hs

```diff
@@ -164,133 +162,121 @@ proxyScript = plam $ \authSymbol' ctx -> P.do
-            let singleScriptInput =
-                  ptraceInfoIfFalse "Transaction must not include script inputs other than own input." $
-                    (plength # scriptInputs) #== 1
+                    txCerts #== pconstant []
 
             foldr1
               (#&&)
-              [ singleAuthTokenBurned
-              , singleOutputWithDatum
+              [ singleOutputWithDatum
               , noCertsIncluded
-              , singleScriptInput
-              , singleGat3Minted
-              , onlyGatsMinted
+              , mintCheck
               ]
           PMintingScript currencySymbol -> P.do
+            -- TODO: punsafeCoerce this
             let ownCurrencySymbol = pfromData currencySymbol
             PCurrencySymbol rawScriptHash <- pmatch ownCurrencySymbol
             let ownScriptHash = pcon $ PScriptHash rawScriptHash
 
-            let proxyValidatorInputs =
-                  pfilter
-                    # plam
-                      ( \input -> pmatch input $ \case
-                          PTxInInfo _ resolved -> P.do
-                            PTxOut addr _ _ _ <- pmatch resolved
-                            PAddress cred _ <- pmatch addr
-                            pmatch cred $ \case
-                              PScriptCredential scriptHash ->
-                                pfromData scriptHash #== ownScriptHash
-                              _ -> pcon PFalse
-                      )
-                    # inputs
+            let isValidatorInput =
+                  plam
+                    ( \input -> pmatch (pfromData input) $ \case
+                        PTxInInfo _ resolved -> P.do
+                          PTxOut addr _ _ _ <- pmatch resolved
+                          PAddress cred _ <- pmatch addr
+                          pmatch cred $ \case
+                            PScriptCredential scriptHash ->
+                              pfromData scriptHash #== ownScriptHash
+                            _ -> pcon PFalse
+                    )
 
             -- Spending Condition 1: Transaction contains an input from Proxy Spending Validator.
             let hasProxyValidatorInput =
                   ptraceInfoIfFalse "Transaction must contain an input from Proxy Validator." $
-                    (plength # proxyValidatorInputs) #== 1
-
-            PValue mintAssetMap <- pmatch mint
-
-            let assets = AssocMap.ptryLookup # ownCurrencySymbol # mintAssetMap
-            PMap assetList <- pmatch assets
-            let assetCount = plength # assetList
-
-            let mintedGAT3 = AssocMap.ptryLookup # padaToken # assets
+                    (pcountIf # isValidatorInput # pfromData inputs) #== 1
 
             -- Spending Condition 2: Transaction mints only one token with currency symbol equal to own hash.
             -- Spending Condition 3: Minted token has empty token name.
```

Comment:
I wonder if these conditions are necessary actually. Minting context already forces Spending part to be ran and it does exactly this check already

Reply:
I agree, it should be safe to remove.